services/data/src/react/hooks/useDataQuery.ts

```diff
@@ -70,8 +110,9 @@ export const useDataQuery = (
      */
 
     const engine = useDataEngine()
-    const queryKey = [staticQuery, variables]
-    const queryFn = () => engine.query(staticQuery, { variables })
+    const queryKey = [staticQuery, queryState.current.variables]
```

Comment:
I guess we could because of how we've defined the queryFn on line 114. Though technically our usage of the queryFn is a bit wrong. To make sure there's no misalignment between the queryKey and the queryFn, the queryFn receives the queryKey and uses that to construct the query.

We're not doing that here and maybe we should, because it's a little unconventional. But, if we don't do that then we could do what you're suggesting.

Reply:
Maybe I'm reading too much into the queryFn semantics though. You could also assume that it's just so that it can be defined elsewhere. I'm not sure of the exact intentions.

Reply:
Ah yeah, I was also thinking that it's a bit odd to recreate the queryFn on every render (with different closure scope) - agree it's probably nicer to take the object as input to the queryFn so we should leave the cache key as-is

Reply:
Resolving, since this isn't something we need to do now (or maybe ever)